tile_cache.go

```diff
@@ -0,0 +1,45 @@
+package sm
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/Wessie/appdirs"
+)
+
+type TileCache interface {
+	// Root path to store cached tiles in with no trailing slash.
+	Path() string
+	// Permission to set when creating missing cache directories.
+	Perm() os.FileMode
+}
```

Comment:
I wrote this as an interface because this would allow me to write my own implementation that I can pass into `Context.SetCache` without changing the API here.